core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -442,6 +443,11 @@ public Path getWorkingDirectory() {
    * Hadoop compatible API and native Alluxio API in the same process.
    *
    * If hadoop file system cache is enabled, this method should only be called when switching user.
+   *
+   * Connection configuration is based on the following order with decreasing priority:
+   *  (1) Zookeeper configuration in Alluxio on Zookeeper URI (Source == URI)
+   *  (2) Zookeeper configuration in configuration files (Source == RUNTIME)
+   *  (3) master hostname and port in Alluxio URI (Source == URI)
```

Comment:
This is a good question.
For the source, since I make URI the highest priority, The URIs should always be the highest priority.

The initial thought was to maintain the previous way: if users set zookeeper configuration in files, they can pass in whatever hostname and port in the URI.

Will change to put the URIs always in the highest priority for HDFS API.

Reply:
Good point about the backward compatibility break. I think the story is much better if we consistently follow URI > conf, but this could break old workflows which set the URI to something different from conf. @apc999 what are your thoughts? I will support either approach

Reply:
As discussed with @apc999, URI is favored by the customers to always have the highest priority.
We could support URI > conf in HDFS first and native API in the future. Thanks!